firedrake/solving.py

```diff
@@ -52,6 +52,14 @@ def solve(*args, **kwargs):
     :class:`.EquationBC`\s specifying, respectively, the strong boundary conditions
     to apply and PDEs to solve on the boundaries.
     For the format of `solver_parameters` see below.
+    
+    Optionally, an argument `P` of type :class:`~.MatrixBase` can be passed to
+    construct any preconditioner from; if none is supplied `A` is used to
+    construct the preconditioner.
```

Comment:
```suggestion
    Optionally, an argument ``P`` of type :class:`~.MatrixBase` can be passed to
    construct any preconditioner from; if none is supplied ``A`` is used to
    construct the preconditioner.
```

I think. We should check that the docstrings render correctly from the documentation CI artefact.